src/htmx.js

```diff
@@ -1280,29 +1294,33 @@ return (function () {
                                     triggerSpec.delay = parseInterval(consumeUntil(tokens, WHITESPACE_OR_COMMA));
                                 } else if (token === "from" && tokens[0] === ":") {
                                     tokens.shift();
-                                    var from_arg = consumeUntil(tokens, WHITESPACE_OR_COMMA);
-                                    if (from_arg === "closest" || from_arg === "find" || from_arg === "next" || from_arg === "previous") {
-                                        tokens.shift();
-                                        var selector = consumeUntil(
-                                            tokens,
-                                            WHITESPACE_OR_COMMA
-                                        )
-                                        // `next` and `previous` allow a selector-less syntax
-                                        if (selector.length > 0) {
-                                            from_arg += " " + selector;
+                                    if (COMBINED_SELECTOR_START.test(tokens[0])) {
+                                        var from_arg = consumeCSSSelector(tokens);
+                                    } else {
+                                        var from_arg = consumeUntil(tokens, WHITESPACE_OR_COMMA);
```

Comment:
Am I missing something, or is this `if` construct redundant, since `consumeCSSSelector` already has a `if` construct where if checks for the start of the combined selector?

Reply:
I believe it is necessary, to differentiate between `from:previous (#a #b)` and `from:(#a #b)`; in the prior case, if we don't have this check and immediately jump to the block in the else statement, we'll first extract "previous" as a CSS selector, followed by a htmx syntax error being thrown due to the remaining `(#a #b)`

It'd be more readable if the condition of that `if` was "if one of the keywords (closest, find...) is there", but we only know that once we've grabbed then next token, but to grab the next token we first have to check if it is a descendant combinator selector, as those parse differently than other keywords.

Reply:
Scratch my previous comment, you're entirely correct.

Reply:
That test has to be there to make f.i. `hx-trigger="click from:(#a #b)"` work. If you remove the `if`-part and jump straight to the `else`-block, it'll only work if you actually use a keyword like `previous` etc..

Basically the code reads "if it's a descendant combinator, parse it, otherwise, check if it's either a simple selector or a keyword with a simple/descendant combinator and parse accordingly".

Your comment dit make me realize that the syntax of `from:<descendant combinator>` wasn't covered by the tests, so I improved them a little.